quinn/src/endpoint.rs

```diff
@@ -47,6 +47,14 @@ pub struct Endpoint {
     runtime: Arc<dyn Runtime>,
 }
 
+pub struct EndpointStats {
+    open_connections: usize,
+    incoming_handshakes_since_last_sample: usize,
+    total_incoming_handshakes: usize,
+    total_outgoing_handshakes: usize,
+    outgoing_handshakes_since_last_sample: usize,
```

Comment:
`_since_last_sample` counters are redundant: a caller interested in progression over time can store the previous total and compute changes. This both simpler for us to implement and less lossy, because a caller can aggregate over any time interval. Similar logic probably applies to `open_connections` too.